src/operator/subgraph/dnnl/dnnl_identity_property.h

```diff
@@ -65,10 +66,11 @@ class SgDNNLIdentitySelector : public SubgraphSelectorV2 {
   }
 
   bool SelectInput(const BiDirectedNode& n, const BiDirectedNode& input_node) override {
-    if (input_node.node->is_variable()) {
+    if (pattern_found || input_node.node->is_variable()) {
```

Comment:
You could use `matched_list_` like you did in the `Filter` function.
```suggestion
    if (matched_list_.size() == 2 || input_node.node->is_variable()) {
```
If not, then I would also use `pattern_found` in the `Filter` function, to show, that these conditions are connected.
https://github.com/apache/incubator-mxnet/blob/7e163157cf789d7ec5e7272532e04495e57bd0ba/src/operator/subgraph/dnnl/dnnl_identity_property.h#L85
Change to :
```
    if (pattern_found && candidates.size() == matched_list_.size()) { 
```

Reply:
Done